v2/tools/generator/internal/codegen/pipeline/state.go

```diff
@@ -131,10 +109,43 @@ func (s *State) CheckFinalState() error {
 // copy creates a new independent copy of the state
 func (s *State) copy() *State {
 	return &State{
-		definitions:        s.definitions.Copy(),
-		conversionGraph:    s.conversionGraph,
-		exportedConfigMaps: s.exportedConfigMaps.Copy(),
-		stagesSeen:         s.stagesSeen,
-		stagesExpected:     s.stagesExpected,
+		definitions:    s.definitions.Copy(),
+		stagesSeen:     s.stagesSeen,
+		stagesExpected: s.stagesExpected,
+		stateInfo:      maps.Clone(s.stateInfo),
+	}
+}
+
+// StateWithInfo returns a new state with the given information included.
+// Any existing information with the same key is replaced.
+// Has to be written as a standalone function because methods can't introduce new generic variation.
+func StateWithInfo[I any](
```

Comment:
minor, style: prefer `T` for generic type variable, rather than I?

Reply:
I used `I` for info, mirroring the way the Golang libraries use single capital letters, e.g. the `maps` package uses `K` for the type of key, `V` for the type of value.